### PR TITLE
precipitation interppolation vignette 

### PR DESCRIPTION
**Pull request by spencerrhea**
Here is the precipitation interpolation vignette. In addition, ms_conversations checks macrosheds_root before downloaded ms_vars. A few tests are fixed to reflect other changes. 

One thing to note. I added a data folder for package data. I wanted to include a watershed boundary in the package so the vignette could get a watershed, to use in the precip interp function, without additional steps. I don't know if there is a down side to having package data, other than in some tests or other areas saving things to a data folder and that could be confusing (tho I think I git ignored everything in the folder without a negative {!thing}). 

@vlahm

### CODE REVIEW

**Review by vlahm — CHANGES_REQUESTED** [review]
@spencerrhea I love the idea of shipping a watershed boundary with the package. That's pro. 

I made a few comments on little things throughout.

File: R/ms_conversions.R [hunk 1]
```diff
@@ -26,6 +26,10 @@
 #'    converted according to the atomic mass of their main constituent. For example,
 #'    this can be used to convert NO3-N to NO3, or NO3 to NO3-N. See details.
 #'    [ms_download_variables()]. If not supplied, ms_vars will be retrieved automatically.
+#' @param macrosheds_root file path. Path to a folder where macrosheds data is stored. 
+#'    This path is needed because ms_conversions needs information on 
+#'    units in order to perform conversions. If the ms_vars file is not in this directory 
+#'    then the file will be downloaded and saved to in the folder as ms_vars.feather..
```
[the remark this answers is not in the document]
Comment (spencerrhea) [reply R/ms_conversions.R]: Yep!

File: R/ms_calc_watershed_precip.R [hunk 2]
```diff
@@ -426,7 +426,7 @@ ms_calc_watershed_precip <- function(precip,
                     j = 1:min(nthreads, nrow(precip_superchunk)),
                     .combine = idw_parallel_combine,
                     .init = 'first iter',
-                    .packages = c('dplyr', 'raster')) %parcond% {
+                    .packages = c('magrittr', 'dplyr', 'raster')) %parcond% {
```
[the remark this answers is not in the document]
Comment (spencerrhea) [reply R/ms_calc_watershed_precip.R]: Another great catch! It indeed does not need magrittr.

File: R/ms_internals.R [hunk 3]
```diff
@@ -714,12 +714,12 @@ ms_parallelize <- function(maxcores = Inf){
     # clst <- parallel::makeCluster(ncores)
     clst <- NULL
     ncores <- min(parallel::detectCores(), maxcores)
+    clst <- parallel::makeCluster(ncores)
```
[the remark this answers is not in the document]
Comment (spencerrhea) [reply R/ms_internals.R]: Another good one!

**Comment by spencerrhea** [comment]
@vlahm Thanks for all these catches; I think they are all resolved!

**Review by vlahm — APPROVED** [review]
Right-o! Thank you

**Pull request merged**